vdm-io/Joomla-Component-Builder
# Library manager implementation

Pull Request for many Issues

### Summary of Changes
- Added interface of the new decoupled libraries concept gh-92 
- String Manipulation to SimpleXML and DOMDocument (thanks to @stutteringp0et)
-  Improved the get snippets area to load one library at a time. started on the adaptation of the compiler that is needed for the new libraries
- Resolve gh-111Implement onContentPrepare (thanks to @stutteringp0et)
- Renamed the uikit lib folders.
- Resolved the gh-188 by updating the note.
- Resolved gh-92 by implementation of the library manager in the compiler.
- Resolve gh-181 to insure all urls added to xml is converting special characters with htmlspecialchars.

### Testing Instructions
- Test the new library implementation in all related views
- Check the fields xml files created (to test the new SimpleXML and DOMDocument )
- When adding a library to a site view and see if the library details is added to the **view.html.php**

### Expected result
>This pull request is mostly enhancements

### Actual result
>This pull request is mostly enhancements

### Documentation Changes Required
> I will make a few new tutorials once we have this ready and merged to the new release.
